plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -465,6 +462,33 @@ private static Integer getIcebergFieldId(OrcColumn column)
         return Integer.valueOf(icebergId);
     }
 
+    private static Type getOrcReadType(Type columnType, TypeManager typeManager)
+    {
+        if (columnType == UUID) {
+            // ORC spec doesn't have UUID
+            // TODO read into Int128ArrayBlock for better performance when operating on read values
+            // TODO: Validate that the OrcColumn attribute ICEBERG_BINARY_TYPE is equal to "UUID"
```

Comment:
can we validate this for top-level columns (as we used to)?  we could provide `Optional<Map> attributes` that would be non-empty for top-level and we would validate ICEBERG_BINARY_TYPE for top-level only, and have TODO for nested.

for nested columns, do we write ICEBERG_BINARY_TYPE attribute actually?